Improve documentation and user interface quality (#3)
b1f6c1c4/hub-sync: README.md

> Sync your github forks without git in O(1) time, space, and network BW.

## Basic usage

1. Install the package with *npm* ([nodejs package manager](https://nodejs.org/)).

    ```bash
    npm i -g @b1f6c1c4/hub-sync
    ```

1. Update your fork *instantly*.
For the first time you run this, you will be asked to generate a [GitHub Personal Access Token](https://github.com/settings/tokens).
Follow the instructions carefully and **keep your token CONFIDENTIAL.**

    ```bash
    hub-sync <name-of-your-fork-repository>
    ```

## Advanced usage

1. Update a certain branch:

    ```bash
    hub-sync <repo>/<branch>
    ```

1. Update from another user's fork:

    ```bash
    hub-sync <your-repo> <another-user>
    ```

1. Update from another user's fork, but with a different name:

    ```bash
    hub-sync <your-repo> <another-user>/<repo>
    ```

1. Update from another user's fork, but with a different name and a different branch:

    ```bash
    hub-sync <your-repo>/<branch> <another-user>/<repo>/<branch>
    ```

1. Point a branch of your repo to a specific SHA-1: (rarely used)

    ```bash
    hub-sync <your-repo>/<branch> <sha-1>
    ```

1. Create a new branch

    ```bash

>     hub-sync -c ... ...

Author: `...` means that, in this case, you can use whatever user-repo-branch tuple format(s) listed above.

>     hub-sync -c ... ...

Reviewer: How about continuing use BNF format, like this:
```suggestion
    hub-sync -c <destination> <source>
```

    ```

1. Update even if not fast-forward: **(EXTREMELY DANGEROUS)**

    ```bash

>     hub-sync -f ... ...

Author: ```suggestion
    hub-sync -f <destination> <source>
```

    ```

1. Delete a branch: **(EXTREMELY EXTREMELY DANGEROUS)**

    ```bash

>     hub-sync --delete ... ...

Reviewer: I guess it could be like this...
```suggestion
    hub-sync --delete <destination>
```

    ```

1. See `hub-sync --help` for the complete usage documentation.

Notice: There is **NO SAFETY NET** at all for `-f|--force` and `-d|--delete`.

> **YOU MAY LOSE ALL YOUR DATA IMMEDIATELY** if not used properly.

Author: I don't quite think so, could you please elaborate?

> **YOU MAY LOSE ALL YOUR DATA IMMEDIATELY** if not used properly.

Reviewer: Well... I understand all of the information this tool is dealing with is 'data', yet under this context, it's more than the data in the narrow sense. For data science related or machine learning projects, the data refers to the data set commonly, which is distinguished from program or model. So in the scale of a manual for general audience, I guess using 'work' could be better.

> **YOU MAY LOSE ALL YOUR DATA IMMEDIATELY** if not used properly.

Author: Maybe the following is better?
```suggestion
**YOU MAY LOSE ALL YOUR WORK ON THAT BRANCH IMMEDIATELY** if not used properly.
```

> **YOU MAY LOSE ALL YOUR DATA IMMEDIATELY** if not used properly.

Reviewer: LGTM.
